src/services/EdFi.DataManagementService.Api/Core/Pipeline/PipelineContext.cs

```diff
@@ -0,0 +1,50 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using EdFi.DataManagementService.Api.ApiSchema;
+using EdFi.DataManagementService.Api.Core.Model;
+
+namespace EdFi.DataManagementService.Core.Pipeline;
+
+/// <summary>
+/// The full context of the request, enriched by pipeline steps and the final handler
+/// </summary>
+public class PipelineContext(FrontendRequest _frontendRequest)
```

Comment:
Except for FrontendRequest, PipelineContext is intended to be mutable as passed down the pipeline.